packages/datetime/src/datePicker.tsx

```diff
@@ -264,6 +283,13 @@ export class DatePicker extends AbstractPureComponent<IDatePickerProps, IDatePic
         this.updateValue(value, true);
     };
 
+    private handleTimeChange = (time: Date) => {
+        Utils.safeInvoke(this.props.timePickerProps.onChange, time);
+        const { value } = this.state;
```

Comment:
is it theoretically possible that state.value would be changed as a result of invoking `this.props.timePickerProps.onChange` ? Also what if `timePickerProps === null` ???

Reply:
`null` is not a valid value. it's `X | undefined`